### PR TITLE
Fix pyyaml version

### LINKED ISSUE #1891: CI is broken
Probably related to the new release of Cython: 3.0.0. See [this](https://github.com/yaml/pyyaml/issues/724).

### PR DESCRIPTION
**Pull request by ocelotl**
This is related to the recent release of Cython 3.0.0.

See https://github.com/yaml/pyyaml/issues/724

Fixes #1891

### CODE REVIEW

**Review by srikanthccv — APPROVED** [review]
Do you think this should be changed back in follow-up when things are sorted from upstream deps?

File: tox.ini [hunk 1]
```diff
@@ -502,7 +502,7 @@ commands =
   codespell
 
 [testenv:lint]
-basepython: python3.10
+basepython: python3.9
```
Comment (lzchen) [inline tox.ini:505]: Is this change needed?

**Review by lzchen — APPROVED** [review]

File: tox.ini (hunk 1 above)
Comment (ocelotl) [reply tox.ini:505]: Yes, seems related, `lint` fails without this change

**Comment by ocelotl** [comment]
> Do you think this should be changed back in follow-up when things are sorted from upstream deps?

Right, opened #1893
